DeviceSimulation/SimulationTests.cs

```diff
@@ -29,5 +40,190 @@ public void Should_Return_OK_Status()
             // Assert
             Assert.Equal(HttpStatusCode.OK, response.StatusCode);
         }
+
+        [Fact, Trait("Type", "IntegrationTest")]
+        public void Should_Return_Currently_Running_Simulation()
+        {
+            var request = new HttpRequest(DS_ADDRESS + "/simulations/1");
+            var response = this.httpClient.GetAsync(request).Result;
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+        }
+
+        [Fact, Trait("Type", "IntegrationTest")]
+        public void Should_Create_Default_Simulation()
+        {
+            //delete the simulation that is already running
+            this.Should_Delete_Existing_Simulation();
+            var simulation = JObject.Parse(@"{   
+                'Enabled': true, 
+                'DeviceModels': [   
+                    {   
+                        'Id': 'truck-01', 
+                        'Count': 5 
+                    } 
+                ]
+            }");
+            var request = new HttpRequest(DS_ADDRESS + "/simulations");
+            request.AddHeader("Content-Type", "application/json");
+            request.SetContent(simulation);
+            var response = this.httpClient.PostAsync(request).Result;
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+        }
+
+        [Fact, Trait("Type", "IntegrationTest")]
+        public async void Should_Create_SimulatedDevice()
+        {
+            //delete the simulation that is already running
+            this.Should_Delete_Existing_Simulation();
+            var simulation = JObject.Parse(@"{   
+                'Enabled': true, 
+                'DeviceModels': [   
+                    {   
+                        'Id': 'truck-01', 
+                        'Count': 5 
+                    } 
+                ]
+            }");
+            var request = new HttpRequest(DS_ADDRESS + "/simulations");
+            request.AddHeader("Content-Type", "application/json");
+            request.SetContent(simulation);
+            var response = this.httpClient.PostAsync(request).Result;
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+            Thread.Sleep(15000);
+            RegistryManager registry = RegistryManager.CreateFromConnectionString(IOTHUB_CONNECTION_STRING);
+            Twin dev1 = await registry.GetTwinAsync("truck-01.0");
+            Assert.True(dev1.Tags["IsSimulated"] == 'Y');
```

Comment:
What will happen if the device creation failed? dev1 will be null?

Reply:
yes, dev1 will be null if device creation failed. will add a null check on dev1